test/setup-beam.test.js

```diff
@@ -196,6 +196,11 @@ async function testOTPVersions() {
     expected = 'master'
     got = await setupBeam.getOTPVersion(spec, osVersion, hexMirrors)
     assert.deepStrictEqual(got, expected)
+
+    spec = 'latest'
+    osVersion = 'ubuntu-22.04'
+    got = await setupBeam.getOTPVersion(spec, osVersion, hexMirrors)
+    assert.match(got, /[.]/g)
```

Comment:
Does this simply test you "got something"? Doesn't sound very robust. 😄 

In this "suite" you should find a list of versions that, wihout trying to download the artefacts check the output versus a pre-defined list: https://github.com/erlef/setup-beam/blob/main/test/setup-beam.test.js#L357

It'd be preferable to have the tests there, since the download mechanism is already well tested.

Reply:
That test, is testing a function which is generic and to a degree that makes little sense to the project since some of the values are 1, 1.1 and etc. Those versions do not exist to my knowledge in any of the supported projects. As mentioned earlier it would require some heavy refactoring to that function, to get it to spit out the latest version, not impossible but the test values should be reconsidered. And ideally we only test valid versions of OTP, Elixir, Gleam, rebar and mix. Testing any potential versions makes little sense, since it would only introduce arbitrary rules which does not reflect reality.

Reply:
I'm Ok if you create a set of versions that are SemVer and/or representative or the truth in the applications we support (Erlang, Elixir, Gleam, `rebar3`). What I don't see expressed, but maybe I missed it, is that the algo. is capable of choosing a pre-defined well-known "latest" from a given set of versions (this is what the tests require).

Reply:
Cool. This was a bit of a learning experience for me, but I think it ticks all the boxes.

Reply:
You went all the way 😄 What I meant was for you to either replicate or add to `https://github.com/erlef/setup-beam/blob/main/test/setup-beam.test.js#L357` 😄. You know the versions, so you know what your function should return; it'd be a minimal unit test (like the ones we already have). I mean, taking it from the source is also Ok, and it should be easy to keep those files updated, assuming you just copy-pasted them from their source (?)

Reply:
Fwiw: I really enjoy the latest straightforward and simple implementation.

Reply:
> 1, 1.1 and etc. Those versions do not exist to my knowledge in any of the supported projects

The versions themselves do not exist as releases, but `setup-beam` already infers the latest in a given "version branch". e.g. `24` could resolve to `24.3.4.17` so they make sense in the current implementation.